src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Context.cs

```diff
@@ -341,6 +341,14 @@ public bool Validate(string userName, string password, ContextOptions connection
                 return (BindSam(_serverName, userName, password));
             }
         }
+
+        public void Dispose()
+        {
+            foreach (LdapConnection connection in _connCache.Values)
+            {
+                connection.Dispose();
+            }
+        }
```

Comment:
In the other methods of `CredentialValidator`, access to `_connCache` is guarded by locks. However, I'm not sure what the guidelines with respect to multithreading in `IDisposable.Dispose()` methods are. I assumed that `Dispose()` is only supposed to be called when the object is definitely not used anymore (by any thread), so adding a lock may not be necessary. If this is not in line with the guildelines, feel free to tell me and I will update the PR.